spyder/plugins/onlinehelp/plugin.py

```diff
@@ -10,90 +10,85 @@
 import os.path as osp
 
 # Third party imports
-from qtpy.QtWidgets import QVBoxLayout
+from qtpy.QtCore import Signal
 
 # Local imports
-from spyder.config.base import _, get_conf_path
-from spyder.api.plugins import SpyderPluginWidget
-from spyder.py3compat import PY2, to_text_string
+from spyder.api.plugins import Plugins, SpyderDockablePlugin
+from spyder.api.translations import get_translation
+from spyder.config.base import get_conf_path
 from spyder.plugins.onlinehelp.widgets import PydocBrowser
 
+# Localization
+_ = get_translation('spyder')
 
-class OnlineHelp(SpyderPluginWidget):
-    """Online Help Plugin."""
 
-    CONF_SECTION = 'onlinehelp'
+# --- Plugin
+# ----------------------------------------------------------------------------
+class OnlineHelp(SpyderDockablePlugin):
+    """
+    Online Help Plugin.
+    """
+
+    NAME = 'onlinehelp'
+    TABIFY = Plugins.Help
+    CONF_SECTION = NAME
     CONF_FILE = False
-    LOG_PATH = get_conf_path(CONF_SECTION)
+    WIDGET_CLASS = PydocBrowser
+    LOG_PATH = get_conf_path(NAME)
+
+    # --- Signals
+    # ------------------------------------------------------------------------
+    sig_load_finished = Signal()
+    """
+    This signal is emitted to indicate the help page has finished loading.
+    """
+
+    # --- SpyderDockablePlugin API
+    # ------------------------------------------------------------------------
+    def get_name(self):
+        return _('Online help')
+
+    def get_description(self):
+        return _(
+            'Browse and search the currently installed modules interactively.')
 
-    def __init__(self, parent):
-        SpyderPluginWidget.__init__(self, parent)
+    def get_icon(self):
+        return self.create_icon('help')
 
-        self.pydocbrowser = PydocBrowser(self, self.options_button)
+    def on_close(self, cancelable=False):
+        self.save_history()
+        self.set_conf_option('zoom_factor',
+                             self.get_widget().get_zoom_factor())
+        return True
 
-        layout = QVBoxLayout()
-        layout.addWidget(self.pydocbrowser)
-        self.setLayout(layout)
+    def register(self):
+        widget = self.get_widget()
+        widget.load_history(self.load_history())
+        widget.sig_load_finished.connect(self.sig_load_finished)

```

Comment:
Maybe changing this line for something like `widget.sig_load_finished.connect(self.sig_load_finished.emit)` could fix the failing test @goanpeca ?

Reply:
>Maybe changing this line for something like widget.sig_load_finished.connect(self.sig_load_finished.emit) could fix the failing test @goanpeca ?

It passes locally on my mac, and those tests were failing before this PR and were skipped on Linux and Mac. Also signals should not normally need to have that in place, they coual automatically emit if connected as pointed out. But with what you suggest I will give a try to

`widget.sig_load_finished.connect(lambda: self.sig_load_finished())`  however, it seems more likely that something is blocking the server from starting, maybe the port :shrug: ?

Anyway, now test pass on Linux and win, which is better than the case before :-)